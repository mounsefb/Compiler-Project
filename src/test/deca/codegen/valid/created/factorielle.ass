	TSTO #9
	BOV stack_overflow_error
	ADDSP #22
; start main program
; Main program
; Beginning of main instructions:
	LOAD #1, R2
	STORE R2, 2(GB)
	LOAD #1, R2
	STORE R2, 3(GB)
	LOAD #1, R2
	STORE R2, 4(GB)
Debut_while1:
	LOAD 2(GB), R2
	CMP #0, R2
	BEQ Fin_while1
	LOAD #7, R2
	STORE R2, 1(GB)
	LOAD 1(GB), R2
	LOAD #0, R3
	CMP R3, R2
	BLT lower3
	LOAD #0, R2
	BRA notLower3
lower3:
	LOAD #1, R2
notLower3:
	CMP #1, R2
	BNE else3
then3:
	WSTR "Erreur "
	WNL
	WSTR "Re entrer un entier non nul"
	BRA end_If3
else3:
	LOAD #0, R2
	STORE R2, 2(GB)
Debut_while4:
	LOAD 4(GB), R2
	LOAD 1(GB), R3
	CMP R3, R2
	BLE lower_or_equal5
	LOAD #0, R2
	BRA not_lower_or_eq5
lower_or_equal5:
	LOAD #1, R2
not_lower_or_eq5:
	CMP #0, R2
	BEQ Fin_while4
	LOAD 3(GB), R2
	LOAD 4(GB), R3
	MUL R3, R2
	STORE R2, 3(GB)
	LOAD 4(GB), R2
	LOAD #1, R3
	ADD R3, R2
	STORE R2, 4(GB)
	BRA Debut_while4
Fin_while4:
	WSTR "Le factorielle de "
	LOAD 1(GB), R1
	WINT
	WSTR " = "
	LOAD 3(GB), R1
	WINT
	WNL
end_If3:
	BRA Debut_while1
Fin_while1:
	HALT
; end main program
overflow_error:
	WSTR "Error: Overflow during arithmetic operation"
	WNL
	ERROR
stack_overflow_error:
	WSTR "Error: Stack Overflow "
	WNL
	ERROR
io_error:
	WSTR "Error: Input/Output error"
	WNL
	ERROR
Dereferencement.null:
	WSTR "Erreur : dereferencement de null"
	WNL
	ERROR
pile_pleine:
	WSTR "Erreur : pile pleine"
	WNL
	ERROR
tas_plein:
	WSTR "Erreur : allocation impossible, tas plein"
	WNL
	ERROR
cast_incorrect:
	WSTR "la conversion (cast) est incorrecte"
	WNL
	ERROR
; code de l'initialisation des champs
; code des m√©thodes
code.object.equals:
	LOAD -2(LB), R0
	CMP -3(LB), R0
	BEQ if_equal_pour_object.equals
	LOAD #0, R0
	BRA fin_methodes
if_equal_pour_object.equals:
	LOAD #1, R0
	BRA fin_methodes
fin_methodes:
	RTS
