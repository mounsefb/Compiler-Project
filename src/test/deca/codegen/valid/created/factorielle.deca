// Description :
//    Test de l'affectation d'une variable a Ã  une variable b 
//     On a tester avec le readInt() ca marche maintenant on el force avec 43 pour passer les test auto 
//
// Resultats :
//    Ok
{
    
    int n;
    boolean badInt = true;
    int res = 1;
    int i =1;
    while(badInt){
        n = 7;
        if( n < 0){
            println("Erreur ");
            print("Re entrer un entier non nul");
        }
        else{
            badInt = false;
            while(i<=n){
                res = res * i;
                i = i + 1;
            }
            println("Le factorielle de ", n , " = ",res);

        }
    }
}