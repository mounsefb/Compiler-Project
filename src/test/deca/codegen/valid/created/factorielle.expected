Le factorielle de 7 = 5040
